INTRODUCTION.md

```diff
@@ -1435,10 +1436,41 @@ To read more about static group membership, see [KIP-345](https://cwiki.apache.o
 
 ### Topics
 
+#### Unknown or unauthorized topics
+
+If a consumer application subscribes to non-existent or unauthorized topics
+a consumer error will be propagated for each unavailable topic with the
+error code set to either `RD_KAFKA_RESP_ERR_UNKNOWN_TOPIC_OR_PART` or a
+broker-specific error code, such as
+`RD_KAFKA_RESP_ERR_TOPIC_AUTHORIZATION_FAILED`.
```

Comment:
99% of the time, users will be subscribing to a topic they know to exist, and they will want their application to fail if it doesn't. in C# for example, I just want to do `await consumer.SubscribeAsync("my-topic");`, and get an exception if there's one of the two problem listed above. setting an error handler is much less ergonomic. remind me again why we can't provide an API for the common case that behaves in this way? There could be an optional flag that makes it work in the way documented here for the 1% who need it.

Reply:
(not something for this PR, but i'll take any opportunity to get up on that soap box)

Reply:
The problem is that subscribe() is a one-time thing (for most applications), while these type of topic errors can happen at any time (even with a synchronous subscribe), which means that the user would have to have error handling for unavailable topics in two places: around the subscribe() call, and during consumption.
I don't think this makes the user program or experience less complex, it is better to educate and make the streaming event consumption as straight-forward as possible, since that is the business we're in.

Reply:
My three responses to that are: 

1. In practice, both of the two examples cited above are going to happen infrequently enough that people aren't going to bother explicitly handling them in the error handler (and I don't think we should recommend that they do). In practice, they will rely on just logging the problem along with other errors on looking at the log when something breaks.

2. Having a synchronous subscribe method throw an exception almost gives you error handling for free, since you'll usually be able to rely on an outer exception handler that you already have in place. It will also be more quickly evident that something when wrong and what the problem was via this mechanism.

3. These problems are most likely to occur on subscribe, and we absolutely add value if we help people work out more easily what a problem was in the majority of cases, even if it doesn't cover them all.

The same sort of argument applies for why we should have an option for synchronous client creation.

in summary, it is better to educate and make the streaming event consumption as straight-forward as possible, since that is the business we're in.